streams-contrib/streams-persist-graph/src/main/java/org/apache/streams/graph/GraphHttpPersistWriter.java

```diff
@@ -84,54 +86,79 @@ else if( configuration.getType().equals(GraphHttpConfiguration.Type.REXSTER)) {
     protected ObjectNode preparePayload(StreamsDatum entry) {
 
         Activity activity = null;
+        ActivityObject activityObject = null;
 
         if (entry.getDocument() instanceof Activity) {
             activity = (Activity) entry.getDocument();
-        } else if (entry.getDocument() instanceof ObjectNode) {
-            activity = mapper.convertValue(entry.getDocument(), Activity.class);
-        } else if (entry.getDocument() instanceof String) {
-            try {
-                activity = mapper.readValue((String) entry.getDocument(), Activity.class);
-            } catch (Throwable e) {
-                LOGGER.warn(e.getMessage());
+        } else if (entry.getDocument() instanceof ActivityObject) {
+            activityObject = (ActivityObject) entry.getDocument();
+        } else {
+            ObjectNode objectNode;
+            if (entry.getDocument() instanceof ObjectNode) {
+                objectNode = (ObjectNode) entry.getDocument();
+            } else if( entry.getDocument() instanceof String) {
+                try {
+                    objectNode = mapper.readValue((String) entry.getDocument(), ObjectNode.class);
+                } catch (IOException e) {
+                    LOGGER.error("Can't handle input: ", entry);
+                    return mapper.createObjectNode();
```

Comment:
Is eating the exception and returning a default object node the right strategy here? Shouldn't the IOE be thrown up to the caller?